### PR TITLE
Update Core to v0.2.7

### PR DESCRIPTION
**Pull request by pjanotti**
**Description:** Update core to v0.2.7 and where appropriate use the new obsreport package. Already, ran `make gotidy`

**Testing:** Manual validation besides local build.

### CODE REVIEW

File: exporter/azuremonitorexporter/traceexporter.go [hunk 1]
```diff
@@ -713,9 +713,7 @@ func newTraceExporter(config *Config, transportChannel transportChannel, logger
 
 	exp, err := exporterhelper.NewTraceExporter(
 		config,
-		exporter.pushTraceData,
-		exporterhelper.WithTracing(true),
-		exporterhelper.WithMetrics(true))
```
Comment (owais) [inline exporter/azuremonitorexporter/traceexporter.go]: Q: we don't need these anymore? Are they automatically enabled by the new observability package?

File: receiver/sapmreceiver/trace_receiver.go [hunk 2]
```diff
@@ -75,36 +74,37 @@ func (sr *sapmReceiver) handleRequest(ctx context.Context, req *http.Request) er
...
+	ctx = obsreport.StartTraceDataReceiveOp(ctx, sr.config.Name(), "http")
+	numSpans := 0
+
 	// process sapm batches
 	for _, batch := range sapm.Batches {
+
 		// convert the jager batches to OCProto
-		td, err := jaegertranslator.ProtoBatchToOCProto(*batch)
+		// TODO: the translator never returns error change the function signature.
+		td, _ := jaegertranslator.ProtoBatchToOCProto(*batch)
```
Comment (owais) [inline receiver/sapmreceiver/trace_receiver.go]: May be we should keep err handling until the translator changes it's signature?

File: exporter/azuremonitorexporter/traceexporter.go (hunk 1 above)
Comment (pjanotti) [reply exporter/azuremonitorexporter/traceexporter.go]: Correct: there were a few misusages of them disabling metrics or traces when we didn't want them.

File: receiver/sapmreceiver/trace_receiver.go (hunk 2 above)
Comment (pjanotti) [reply receiver/sapmreceiver/trace_receiver.go]: Using this fact we can simplify a bit the code if we keep it we should handle it different to get correct metrics.

**Review by owais — APPROVED** [review]

**Review by tigrannajaryan — APPROVED** [review]
Thanks!

**Pull request merged**